program-runtime/src/loaded_programs.rs

```diff
@@ -416,6 +422,16 @@ impl LoadedProgram {
             && slot >= self.deployment_slot
             && slot < self.effective_slot
     }
+
+    pub fn update_access_slot(&self, slot: Slot) {
+        let _ = self.latest_access_slot.fetch_max(slot, Ordering::Relaxed);
+    }
+
+    pub fn decayed_usage_counter(&self, now: Slot) -> u64 {
+        let last_access = self.latest_access_slot.load(Ordering::Relaxed);
+        let decaying_for = now.saturating_sub(last_access);
```

Comment:
`decaying_for` must be limited by `.min(64)` otherwise the line below might overflow on the bitshift.